src/main/scala/uk/gov/hmrc/versioning/ReleaseVersioning.scala

```diff
@@ -21,58 +21,86 @@ object ReleaseVersioning {
   def calculateNextVersion(
     release: Boolean,
     hotfix: Boolean,
+    releaseCandidate: Boolean,
     maybeGitDescribe: Option[String],
     majorVersion: Int): String =
-    calculateVersion(release, hotfix, maybeGitDescribe, majorVersion) + suffix(release)
+    calculateVersion(release, hotfix, releaseCandidate, maybeGitDescribe, majorVersion) + releaseSuffix(release)
 
   private def calculateVersion(
     release: Boolean,
     hotfix: Boolean,
+    releaseCandidate: Boolean,
     maybeGitDescribe: Option[String],
-    requestedMajorVersion: Int): String = maybeGitDescribe match {
+    requestedMajorVersion: Int
+  ): String = maybeGitDescribe match {
 
     case None if requestedMajorVersion > 0 =>
       throw new IllegalArgumentException(
         s"Invalid majorVersion: $requestedMajorVersion. You cannot request a major version of $requestedMajorVersion if there are no tags in the repository."
       )
 
     case None =>
-      "0.1.0"
+      s"0.1.0${releaseCandidateSuffix(releaseCandidate, None)}"
 
-    case Some(tag(AsInt(major), _, _)) if major != requestedMajorVersion && hotfix =>
-      throw new IllegalArgumentException(
-        s"Invalid majorVersion: $requestedMajorVersion. It is not possible to change the major version as part of a hotfix."
-      )
+    case Some(v) => Version.parse(v) match {
+      case Some(Version(major, _, _, _)) if major != requestedMajorVersion && hotfix =>
+        throw new IllegalArgumentException(
+          s"Invalid majorVersion: $requestedMajorVersion. It is not possible to change the major version as part of a hotfix."
+        )
 
-    case Some(tag(AsInt(major), _, _)) if !validMajorVersion(major, requestedMajorVersion) =>
-      throw new IllegalArgumentException(
-        s"Invalid majorVersion: $requestedMajorVersion. " +
-          s"The accepted values are $major or ${major + 1} based on current git tags."
-      )
+      case Some(Version(major, _, _, _)) if !validMajorVersion(major, requestedMajorVersion) =>
+        throw new IllegalArgumentException(
+          s"Invalid majorVersion: $requestedMajorVersion. " +
+            s"The accepted values are $major or ${major + 1} based on current git tags."
+        )
 
-    case Some(tag(AsInt(major), _, _)) if requestedMajorVersion != major =>
-      s"$requestedMajorVersion.0.0"
+      // was previously a releaseCandidate, but not anymore
+      case Some(Version(major, minor, patch, Some(_))) if !releaseCandidate =>
+        s"$major.$minor.$patch"
 
-    case Some(tag(major, minor, AsInt(patch))) if hotfix =>
-      s"$major.$minor.${patch + 1}"
+      case Some(Version(major, _, _, optRc)) if requestedMajorVersion != major =>
```

Comment:
can we make the order of `requestedMajorVersion != major` consistent between this and line 46 please?